### ISSUE TITLE
False positive for scipy.spatial.cDKTree.data unsubscriptable

### ISSUE THREAD

**Issue by jacklovell** [PyCQA/pylint#3104]
### Steps to reproduce
1. Create a `scipy.spatial.cKDTree` instance.
2. Subscript the `data` attribute.

Based on the Scipy documentation for `cKDTree`:
```python
import numpy as np
from scipy.spatial import cKDTree
x, y = np.mgrid[0:5, 2:8]
tree = cKDTree(np.c_[x.ravel(), y.ravel()])

dd, ii = tree.query([[0, 0], [2.1, 2.9]], k=1)
print(dd, ii)

print(tree.data[ii]) 
```

### Current behavior
Pylint raises E1136 "unsubscriptable-object" when indexing `tree.data` (last line of code example)

### Expected behavior
No such warning: this operation is valid and works without error

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.4 (default, Aug 13 2019, 20:35:49) 
[GCC 7.3.0]

**Comment by hippo91** [comment]
@jacklovell thanks for this. 
`pylint̀` is not able to analyze module which are `C` coded. I think this is the case of `scipy.spatial`. Thus it is necessary to implement a *brain* in `astroid` (the ast parser and inference engine used by `pylint`).
If you are interested in developing such a brain, feel free to send a patch or submit a PR !
